registry/coder/modules/amazon-q/amazon-q.tftest.hcl

```diff
@@ -0,0 +1,144 @@
+run "required_variables" {
```

Comment:
This is good, I think it is probably a better approach to testing on this code to use the native Terraform testing framework for unit tests.

2 Items for this
* As this is new for the Coder team, can you put some light documentation in the README about how to contribute and execute the tests. Link to the official documentation predominantly and feed free to steal some of the language from our ADRs

* For these tests, can you pin versions, etc to key versions, such as AgentAPI and Q - We want to certify this module with a specific configuration of AgentAPI and Q and know what breaks when things change from other contributors.